util/statistics.h

```diff
@@ -67,9 +71,7 @@ class StatisticsImpl : public Statistics {
   __declspec(align(64))
   Ticker tickers_[INTERNAL_TICKER_ENUM_MAX]
      __attribute__((aligned(64)));
-  __declspec(align(64))
-  HistogramImpl histograms_[INTERNAL_HISTOGRAM_ENUM_MAX]
-      __attribute__((aligned(64)));
+  std::vector<std::unique_ptr<Histogram>> histograms_;
```

Comment:
I don't feel very comfortable with the array of pointers. How about define a base class

class HistogramCollectionInterface {
  virtual Histogram* GetHistogram(uint32_t histogramType) = 0;
};

and have HistogramCollection and WindowHistogramCollection implement it. Here we put a HistogramCollectionInterface.